pkg/diagnostics/report.go

```diff
@@ -40,4 +55,38 @@ func proxyStatus() {
 			fmt.Printf("	❌ %-12s : not set or empty.\n", proxyName)
 		}
 	}
+	fmt.Println("")
+}
+
+// overriddenEnvVar retrieves and displays overridden environment variables with the prefix `CONFIG_JS_` from the noobaa-core-0 pod
+func OverriddenEnvVar(coreApp *appsv1.StatefulSet, endpointApp *appsv1.Deployment) {
+	log := util.Logger()
+
+	log.Print("⏳ Retrieving overridden environment variable details...\n")
+
+	fmt.Print("Overridden Environment Variables Check (NOOBAA-CORE):\n----------------------------------\n")
+	foundCoreEnv := false
+	for _, envVar := range coreApp.Spec.Template.Spec.Containers[0].Env {
+		if strings.HasPrefix(envVar.Name, "CONFIG_JS_") {
+			fmt.Printf("    	✔ %s : %s\n", envVar.Name, envVar.Value)
+			foundCoreEnv = true
+		}
+	}
+	if !foundCoreEnv {
+		fmt.Printf("	❌ No overridden environment variables found.")
+	}
+	fmt.Println("")
+
+	fmt.Print("Overridden Environment Variables Check (ENDPOINT):\n----------------------------------\n")
+	foundEndpointEnv := false
+	for _, envVar := range endpointApp.Spec.Template.Spec.Containers[0].Env {
+		if strings.HasPrefix(envVar.Name, "CONFIG_JS_") {
+			fmt.Printf("    	✔ %s : %s\n", envVar.Name, envVar.Value)
+			foundEndpointEnv = true
+		}
+	}
+	if !foundEndpointEnv {
+		fmt.Printf("	❌ No overridden environment variables found.")
+	}
```

Comment:
This duplication can be a function that gets a kubernetes object and is called twice, once for core sts and a second time for endpoint deployment.
@achouhan09 Create a new PR that consolidates those into a single function.